langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/protocol/InitializeParams.java

```diff
@@ -0,0 +1,95 @@
+package dev.langchain4j.mcp.client.protocol;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+public class InitializeParams {
```

Comment:
Hmmm... The problem is that I'm using inheritance (`extends McpClientMessage`) which you can't do with records. Not sure how I would replace the inheritance, maybe I just don't know enough about how records work

Reply:
this was a nitpick, no problem at all with using class instead of record

Reply:
https://stackoverflow.com/questions/63605794/is-there-any-way-of-using-records-with-inheritance

> The Java 17 [JLS 8.10](https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.10) notes1 this:
> 
> A record declaration does not have an extends clause, so it is not possible to explicitly declare a direct superclass type, even Record.
> 
> However, a record can implement interfaces, so you can use them polymorphically. Furthermore, since records will inherit any default methods in the interfaces that they implement, they do support a limited form of inheritance.